packages/wrangler/src/pages/deployment-tails.tsx

```diff
@@ -0,0 +1,259 @@
+import { setTimeout } from "node:timers/promises";
+import onExit from "signal-exit";
+import { printWranglerBanner } from "..";
+import { fetchResult } from "../cfetch";
+import { readConfig } from "../config";
+import { getConfigCache } from "../config-cache";
+import { FatalError } from "../errors";
+import { logger } from "../logger";
+import * as metrics from "../metrics";
+import {
+	jsonPrintLogs,
+	prettyPrintLogs,
+	createPagesTail,
+} from "../tail/createTail";
+import { translateCLICommandToFilterMessage } from "../tail/filters";
+import { requireAuth } from "../user";
+import { PAGES_CONFIG_CACHE_FILENAME } from "./constants";
+import { promptSelectProject } from "./prompt-select-project";
+import type { ConfigPath } from "..";
+import type { YargsOptionsToInterface } from "../yargs-types";
+import type { Deployment, PagesConfigCache } from "./types";
+import type { Argv } from "yargs";
+
+type Options = YargsOptionsToInterface<typeof Options> & {
+	// Global flag
+	config?: string;
+};
+const statusChoices = ["ok", "error", "canceled"] as const;
+type StatusChoice = typeof statusChoices[number];
+const isStatusChoiceList = (
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	data?: any[]
+): data is StatusChoice[] =>
+	data?.every((d) => statusChoices.includes(d)) ?? false;
+
+export function Options(yargs: Argv) {
+	return (
+		yargs
+			.positional("deployment-id", {
+				type: "string",
+				description:
+					"Deployment to tail. Specify by environment if " +
+					"deployment ID is unknown.",
+			})
+			.options({
+				"project-name": {
```

Comment:
is this the alternative if no deployment id arg is passed in?

Reply:
project-name is actually always required, but we'll prompt you for it if not specified